### PR TITLE
feat: Sparse checkout

### PR DESCRIPTION
**Pull request by msvticket**
Partial fix for jenkins-x/jx#8240

### CODE REVIEW

**Comment by msvticket** [comment]
@ankitm123 I upgraded to go 1.17 based on you PR

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@ GO_DEPENDENCIES := $(call rwildcard,pkg/,*.go) $(call rwildcard,cmd/j,*.go)
 
 BRANCH     := $(shell git rev-parse --abbrev-ref HEAD 2> /dev/null  || echo 'unknown')
 BUILD_DATE := $(shell date +%Y%m%d-%H:%M:%S)
-CGO_ENABLED = 0
+CGO_ENABLED = 1
```
Comment (ankitm123) [inline Makefile:21]: why?

Comment (msvticket) [reply Makefile:21]: No, but there is a dependency that does. Deep down...

Comment (msvticket) [reply Makefile:21]: So compilation fails without it

Comment (ankitm123) [reply Makefile:21]: Because of the go-upgrade? I will take a look ... cgo opens up hell on earth :grimacing:

**Comment by ankitm123** [comment]
No, I meant with CGO_ENABLED=1, but jx-gitops at 0.3.20, it should pass tests ..

**Comment by ankitm123** [comment]
oh wait, ie-proxy fix does not need CGO, u can do this:
https://github.com/jenkins-x-plugins/jx-pipeline/blob/232704191496244a1afec624cd404128396559a8/go.mod#L187-L188

**Pull request merged**